precompiles/vesting/src/wrapper.rs

```diff
@@ -0,0 +1,100 @@
+// Copyright 2023-2024 Freeverse.io
+// This file is part of LAOS.
+
+// LAOS is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// LAOS is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with LAOS.  If not, see <http://www.gnu.org/licenses/>.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use frame_support::{pallet, traits::tokens::currency::Currency};
+pub use pallet::*;
+use sp_core::{H160, U256};
+use sp_runtime::traits::{Convert, ConvertBack};
+
+#[pallet]
+pub mod pallet {
+
+	use frame_system::pallet_prelude::BlockNumberFor;
+
+	use super::*;
+	use crate::*;
+
+	pub type BalanceOf<Runtime> = <<Runtime as pallet_vesting::Config>::Currency as Currency<
+		<Runtime as frame_system::Config>::AccountId,
+	>>::Balance;
+
+	#[pallet::config(with_default)]
+	pub trait Config: frame_system::Config + pallet_vesting::Config {
+		#[pallet::no_default_bounds]
+		/// Converts `Self::AccountId` to `H160`
+		type AccountIdToH160: ConvertBack<Self::AccountId, H160>;
+
+		#[pallet::no_default_bounds]
+		/// Converts `BalanceOf<Self>` to `U256`
+		type BalanceOfToU256: Convert<BalanceOf<Self>, U256>;
+
+		#[pallet::no_default]
+		/// Converts `BlockNumberFor<Self>` to `U256`
+		type BlockNumberForToU256: Convert<BlockNumberFor<Self>, U256>;
```

Comment:
what about register also default impl for `BlockNumberForToU256` instead of implementing it at runtime level? Same as we did with `AccountIdToH160` and `BalanceOfToU256`?

Reply:
there is a proposal: https://github.com/freeverseio/laos/pull/668

Reply:
PR closed as the code structure has changed and defaults have already been implemented